ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -12,10 +12,17 @@ http {
 {# allow large uploads, need to thread proper limit into here #}
     client_max_body_size 50M;
 
+    map $http_{{ transaction.header | lower | replace('-', '_') }} $ow_tid {
+        # request_id: unique request identifier generated from 16 random bytes, in hexadecimal
+        default $request_id;
+        # If the caller has already set a tid into the header in the correct format, this one will be taken.
+        "~^(?<tid>{{ transaction.regex }})$" $tid;
```

Comment:
The tid is used in the cache logic - if the id isn’t unique (specified by a user, it will affect the cache behavior).